aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -731,14 +731,19 @@ Tensor all(const Tensor& self) {
   TORCH_CHECK(self.layout() == Layout::Strided,
               "all only supports strided layout, got: ", self.layout());
 
-  Tensor result = at::empty({0}, self.options());
-  auto iter = make_reduction(
-    "all", result, self, {}, false, self.scalar_type());
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  if (self.is_cuda()) {
```

Comment:
This needs a comment explaining the CUDA vs CPU discrepancy

Reply:
Done.